Assets/Scripts/Models/Time/TimeManager.cs

```diff
@@ -130,6 +134,16 @@ public void Update(float time)
         // Systems that update every frame.
         InvokeEvent(EveryFrame, time);
 
+        if (nextFrameActions.Count > 0)
+        {
+            for (int i = 0; i < nextFrameActions.Count; i++)
```

Comment:
Personal preference or why do you use a iterative `for` if you could use a `foreach`?